### PR TITLE
Layout window split

### PR DESCRIPTION
**Pull request by makiam**
Refactor class to split huge and complex actionPerformed() into smaller parts. UI tests for Layout window.

### CODE REVIEW

**Comment by Sailsman63** [comment]
I have been considering the `actionPerformed()` method in `LayoutWindow` for a refactor. However, the changes shown here should be two different pull requests.

We need to be able to review the UI tests separately from the refactoring.

File: ArtOfIllusion/src/artofillusion/LayoutWindow.java [hunk 1]
```diff
@@ -15,28 +15,39 @@ Changes copyright (C) 2017 by Petri Ihalainen
...
+import artofillusion.texture.Texture;
 import artofillusion.ui.*;
 import artofillusion.view.ViewAnimation;
-import artofillusion.keystroke.*;
 import buoy.event.*;
 import buoy.widget.*;
-
-import java.awt.*;
+import buoyx.docking.*;
+import java.awt.Color;
```
Comment (Sailsman63) [inline ArtOfIllusion/src/artofillusion/LayoutWindow.java]: Please don't expand package imports like this. I'm aware of the arguments in favor of explicit per-class imports, and in this context, I disagree with them.

```suggestion
import java.awt.*;
```

File: ArtOfIllusion/src/artofillusion/LayoutWindow.java [hunk 2]
```diff
@@ -45,30 +56,31 @@ Changes copyright (C) 2017 by Petri Ihalainen
   SceneViewer theView[];
   BorderContainer viewPanel[];
   FormContainer viewsContainer;
-  FormContainer centerContainer;
+
```
Comment (Sailsman63) [inline ArtOfIllusion/src/artofillusion/LayoutWindow.java]: Removing this breaks the build - it is used in multiple places!

```suggestion
  FormContainer centerContainer;
```

File: ArtOfIllusion/src/artofillusion/LayoutWindow.java [hunk 3]
```diff
@@ -439,59 +450,58 @@ else if ("-".equals(lines[i]))
...
-            item.addEventLink(CommandEvent.class, this, "actionPerformed");
-          }
-      }
+    
+    Collections.sort(translators, Comparator.comparing(Translator::getName));
+    for (Translator translator: translators)
+    {
+        if(translator.canImport())
+        {
+          BMenuItem tim = new BMenuItem(translator.getName());
```
Comment (Sailsman63) [inline ArtOfIllusion/src/artofillusion/LayoutWindow.java]: Who's tim? :)

Seriously, is there a reason that you used this name for the variable?

**Comment by Sailsman63** [comment]
@makiam: Please do not pick changes from master. It makes it harder to keep a clean history.

You can fix this by running `git push -f b062c07`

When the time comes to revisit this, we can rebase on top of what happens to be the current master.

**Comment by makiam** [comment]
Closed this and open new #261

**Pull request closed without merging**